main/src/ca/uwaterloo/flix/language/ast/NamedAst.scala

```diff
@@ -141,7 +141,7 @@ object NamedAst {
 
     case class ArrayLength(base: NamedAst.Expression, loc: SourceLocation) extends NamedAst.Expression
 
-    case class ArraySlice(base: NamedAst.Expression, beginIndex: NamedAst.Expression, endIndex: NamedAst.Expression, loc: SourceLocation) extends NamedAst.Expression
+    case class ArraySlice(region: NamedAst.Expression, base: NamedAst.Expression, beginIndex: NamedAst.Expression, endIndex: NamedAst.Expression, loc: SourceLocation) extends NamedAst.Expression
```

Comment:
I think in a separate PR we should just rename all of these to `exp1`, `exp2`, and so forth we as we usually do.